### PR TITLE
Refactoring of KDIC MainActivity.java

### PR DESCRIPTION
**Pull request by nikagarwal98**
* Added member variables in onDestroy
* Made all member variables private
* Cleaned up setupPlaybackToolbar()

### CODE REVIEW

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 1]
```diff
@@ -468,7 +468,18 @@ public void onBackPressed() {
     @Override
     protected void onDestroy() {
         Log.d(TAG, "MainActivity Destroyed.");
-
         super.onDestroy();
+        super.onDestroy();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Why is this called twice? Also you want to do your clean up before you call `super.onDestroy()`

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 2]
```diff
@@ -468,7 +468,18 @@ public void onBackPressed() {
     @Override
     protected void onDestroy() {
         Log.d(TAG, "MainActivity Destroyed.");
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Logs shouldn't be in production code.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 3]
```diff
@@ -374,13 +380,13 @@ public void showVisualizeFragment() {
...
             animation.setDuration(200);
             animation.setInterpolator(new AccelerateDecelerateInterpolator());
             animation.setFillAfter(true);
-            playPauseButton.setTranslationX(-1 * shiftAmnt);
-            playPauseButton.startAnimation(animation);
+            mPlayPauseButton.setTranslationX(-1 * shiftAmnt);
+            mPlayPauseButton.startAnimation(animation);
 
             // move the info off the screen
             AlphaAnimation alphaAnimation = new AlphaAnimation(1f, 0f);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: What is 1f? A magic number ? Please help us make more meaning of these by extracting them as constants and naming them properly.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 4]
```diff
@@ -374,13 +380,13 @@ public void showVisualizeFragment() {
             mPlaybackToolbar.setNavigationIcon(R.drawable.ic_keyboard_arrow_down_white_24dp);
 
             // move the play button to the middle
-            final float shiftAmnt = (mPlaybackToolbar.getWidth() - playPauseButton.getWidth()) / 2;
+            final float shiftAmnt = (mPlaybackToolbar.getWidth() - mPlayPauseButton.getWidth()) / 2;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Why are we dividing by some random magic number? What is 2 here? Extract as a constant and name properly

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 5]
```diff
@@ -374,13 +380,13 @@ public void showVisualizeFragment() {
...
             TranslateAnimation animation = new TranslateAnimation(shiftAmnt, 0, 0, 0);
             animation.setDuration(200);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Magic number. You know what to do

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 6]
```diff
@@ -374,13 +380,13 @@ public void showVisualizeFragment() {
             mPlaybackToolbar.setNavigationIcon(R.drawable.ic_keyboard_arrow_down_white_24dp);
 
             // move the play button to the middle
-            final float shiftAmnt = (mPlaybackToolbar.getWidth() - playPauseButton.getWidth()) / 2;
+            final float shiftAmnt = (mPlaybackToolbar.getWidth() - mPlayPauseButton.getWidth()) / 2;
             TranslateAnimation animation = new TranslateAnimation(shiftAmnt, 0, 0, 0);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: 0's here do not make sense to anyone reading the code. You might want to extract these as constants and give them descriptive names.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java (hunk 5 above)
Comment (LarryAsante) [reply app/src/main/java/edu/grinnell/kdic/MainActivity.java]: A proper extraction would be `private static final int MS_ANIMATION_DURATION;` (notice the naming convention -- using MS (representing millisecond) to tell the reader that it is time in milliseconds.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 7]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
                             } else {
-                                playPauseButton.setImageResource(R.drawable.ic_loading_spinner);
+                                mPlayPauseButton.setImageResource(R.drawable.ic_loading_spinner);
 
                                 // rotation to use for loading icon
                                 RotateAnimation rotate;
-                                // different center point for rotation if playPauseButton is in the
+                                // different center point for rotation if mPlayPauseButton is in the
                                 // center of the screen
                                 if (mBackStack.peek() != R.id.visualizer)
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Too many unclear numbers here. Extract as constants. See comments below

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 8]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
         mPlaybackToolbar.setNavigationOnClickListener(onToggleVisualizeFragment);
         mPlaybackToolbar.setOnClickListener(onToggleVisualizeFragment);
-        playPauseButton = (ImageView) findViewById(R.id.ib_play_pause);
-        playPauseButton.setOnClickListener(new View.OnClickListener() {
+        mPlayPauseButton = (ImageView) findViewById(R.id.ib_play_pause);
+        mPlayPauseButton.setOnClickListener(togglePlayPauseClickListener());
+    }
+
+    @NonNull
+    private View.OnClickListener togglePlayPauseClickListener() {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:137]: This method is too long. Modularize it.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 9]
```diff
@@ -85,27 +84,19 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            updateSchedule();
-
     }
 
     @Override
     protected void onStart() {
         super.onStart();
         Log.d(TAG, "onStart");
-
         // bind to the radio service
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: No need for this comment

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 10]
```diff
@@ -85,27 +84,19 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         // set the layout to use
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove this comment

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 11]
```diff
@@ -85,27 +84,19 @@ protected void onCreate(Bundle savedInstanceState) {
...
         setContentView(R.layout.activity_main);
-
         setupNavigation(); // setup the nav drawer and navigation functionality
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove these inline comments as well. Use inline comments when you're doing something that absolutely requires explanation

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 12]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
+                if (mRadioService.isPlaying()) {
                     // pause
-                    radioService.pause();
+                    mRadioService.pause();
                     // switch to play icon
-                    playPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
+                    mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
                 } else {
                     // play
                     if (NetworkState.isOnline(MainActivity.this)) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use a static import of this method to make it shorter

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 13]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
+                                mPlayPauseButton.setImageResource(R.drawable.ic_loading_spinner);
 
                                 // rotation to use for loading icon
                                 RotateAnimation rotate;
-                                // different center point for rotation if playPauseButton is in the
+                                // different center point for rotation if mPlayPauseButton is in the
                                 // center of the screen
                                 if (mBackStack.peek() != R.id.visualizer)
                                     rotate = new RotateAnimation(0, 360,
                                             Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF,
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use static imports. Also, handle these strange numbers appropriately by extracting them into meaningfully-named constants. eg. I have no idea what 360 or 0.5f means in this context.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 14]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
                                             0.5f);
                                 else {
-                                    float shiftX = mPlaybackToolbar.getWidth() / -2 + playPauseButton.getWidth();
+                                    float shiftX = mPlaybackToolbar.getWidth() / -2 + mPlayPauseButton.getWidth();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: See comments about numbers above

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 15]
```diff
@@ -114,99 +105,109 @@ protected void onStart() {
...
                                     float shiftY = mPlaybackToolbar.getHeight() / 2;
                                     rotate = new RotateAnimation(0, 360, shiftX, shiftY);
                                 }
                                 rotate.setDuration(1000);
                                 rotate.setRepeatCount(Animation.INFINITE);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use static imports

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 16]
```diff
@@ -264,12 +265,17 @@ public void onClick(View v) {
             }
         });
 
-        // set up backstack
+        // set up mBackStack
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: No need for this comment

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 17]
```diff
@@ -315,7 +321,7 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
                 mDrawerLayout.closeDrawer(GravityCompat.START);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Use static import

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 18]
```diff
@@ -40,7 +41,7 @@
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:45]: Document the class using Javadoc style 
See: https://www.tutorialspoint.com/java/java_documentation.htm

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 19]
```diff
@@ -50,33 +51,31 @@
...
             } else {
-                playPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
+                mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
             }
         }
-
         // Called when the connection with the service disconnects unexpectedly
-        // Changed the boolean boundToRadioService to false.
         public void onServiceDisconnected(ComponentName className) {
             Log.e(TAG, "onServiceDisconnected");
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Logs should be removed

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 20]
```diff
@@ -85,27 +84,19 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        // update the schedule if connected to internet
-        if (NetworkState.isOnline(this))
-            updateSchedule();
-
     }
 
     @Override
     protected void onStart() {
         super.onStart();
         Log.d(TAG, "onStart");
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Remove logs

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Nice start. There are more changes to be done so lets focus on those so we can iterate from there.

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 21]
```diff
@@ -191,6 +168,38 @@ public void run() {
         };
     }
 
+    private void loadingSpinAnimation() {
+        mPlayPauseButton.setImageResource(R.drawable.ic_loading_spinner);
+        // rotation to use for loading icon
+        RotateAnimation rotate;
+        // different center point for rotation if mPlayPauseButton is in the
```
Comment (mattorib) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:175]: Remove these comments

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 22]
```diff
@@ -318,7 +324,7 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
                 }
 
                 // close the drawer after something is clicked
```
Comment (mattorib) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:326]: Remove comment

**Review by mattorib — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 23]
```diff
@@ -74,26 +76,22 @@ public void onServiceConnected(ComponentName className, IBinder service) {
         }
         // Called when the connection with the service disconnects unexpectedly
         public void onServiceDisconnected(ComponentName className) {
-            Log.e(TAG, "onServiceDisconnected");
             mBoundToRadioService = false;
         }
     };
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:84]: Nice! Now, our `onCreate(...)` looks much better!

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 24]
```diff
@@ -481,5 +525,6 @@ protected void onDestroy() {
         mFavoritesFragment = null;
         mRadioService = null;
         mBoundToRadioService = false;
+        super.onDestroy();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Good call!

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 25]
```diff
@@ -261,14 +270,11 @@ private void setupNavigation() {
         mNavigationToolbar.setNavigationOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                mDrawerLayout.openDrawer(GravityCompat.START);
+                mDrawerLayout.openDrawer(START);
             }
         });
 
-        // set up mBackStack
         mBackStack = new Stack<>();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Aha! So you're initializing `mBackStack` here, which is good. But you never free it after use for recycling. You should nullify it in `onDestroy()` else you're likely to leak some memory

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java (hunk 24 above)
Comment (LarryAsante) [reply app/src/main/java/edu/grinnell/kdic/MainActivity.java]: Also: Why are you nullifying only `mFavoritesFragment` and not the others? You should do it for the others!

File: app/src/main/java/edu/grinnell/kdic/MainActivity.java [hunk 26]
```diff
@@ -467,8 +514,17 @@ public void onBackPressed() {
...
-
+        mNavigationToolbar = null;
+        mPlaybackToolbar = null;
+        mPlayPauseButton = null;
+        mDrawerLayout = null;
+        mNavigationView = null;
+        mBackStack = null;
+        mVisualizeFragment = null;
+        mScheduleFragment = null;
+        mFavoritesFragment = null;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/kdic/MainActivity.java:525]: Aight. Sike! You got me on this one! Good job. See my other comments!

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Very nice! See my other comments!

**Pull request closed without merging**